src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/BalanceCalculator.java

```diff
@@ -0,0 +1,167 @@
+package uk.gov.homeoffice.digital.sas.balancecalculator;
+
+import com.google.common.collect.Range;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.homeoffice.digital.sas.balancecalculator.client.RestClient;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.Accrual;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.Agreement;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.Contributions;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.enums.AccrualType;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.timecard.TimeEntry;
+
+@Component
+public class BalanceCalculator {
+
+  private final RestClient restClient;
+
+  @Autowired
+  public BalanceCalculator(RestClient restClient) {
+    this.restClient = restClient;
+  }
+
+  public List<Accrual> calculate(TimeEntry timeEntry) {
+
+    List<Accrual> accrualsToBatchUpdate = Arrays.stream(AccrualType.values())
+        .filter(AccrualType::isEnabled)
+        .map(accrualType -> calculateAccruals(timeEntry, accrualType))
```

Comment:
I'd anticipated this being an array of the classes that know how to perform the calculations.
E.g. `calculators[accrualType].calculate(timeEntry)`
Enabling this to be more dynamic i.e. the only way to extend this for more accrual types at the moment looks to be a switch statement in the calculateAccruals method

Reply:
Actioned